sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -499,7 +500,18 @@ private boolean usesContext(AdditionalParameter param) {
      */
     private Constructor<? extends DoFnInvoker<?, ?>> createInvokerConstructor(
         @SuppressWarnings("rawtypes") Class<? extends DoFnWithContext> clazz) {
+
+      final TypeDescription clazzDescription = new TypeDescription.ForLoadedType(clazz);
+
       DynamicType.Builder<?> builder = new ByteBuddy()
+          // Create subclasses inside the target class's package
+          .with(new NamingStrategy.SuffixingRandom("auxiliary") {
+                @Override
+                public String subclass(TypeDescription.Generic superClass) {
+                  return super.name(clazzDescription);
```

Comment:
Totally fine with being nested inside the other class. As mentioned it may even be better. But please correct the comment (inside the target class rather than inside the target class's package). Maybe also mention that we're doing that so that private/package-protected methods may be accessed?